### DISCUSSION TITLE
set property of custom component required

### DISCUSSION THREAD

**Discussion by AFelahi** [vueform/builder#60] [Questions]
Hi Adam,
is it possible to set a property of a custom component required before save? I tried the following but it didn't work out, example:
`export default class EventStartDateField extends BaseElementField {
	name = "EventStartDateField";

	get schema() {
		return {
			eventStartDate: {
				type: "date",
				label: "Start date",
				extendOptions: {
					allowInput: true,
				},
				attrs: {
					required: true,
				},
				rules:["required"],
				date: true,
				time: true,
			},
		};
	}`

**Comment by adamberecz** [comment]
`DateElement` does not have an `attrs` option. Would `rules: ['required']` not work in your case?
